src/main/scala/io/iohk/ethereum/utils/Config.scala

```diff
@@ -98,6 +99,17 @@ object Config {
       val certificateKeyStorePath: Option[String] = Try(rpcConfig.getString("certificate-keystore-path")).toOption
       val certificatePasswordFile: Option[String] = Try(rpcConfig.getString("certificate-password-file")).toOption
 
+
+      def parseMultipleOrigins(origins: Seq[String]): HttpOriginRange = HttpOriginRange(origins.map(HttpOrigin(_)):_*)
+      def parseSingleOrigin(origin: String): HttpOriginRange = origin match {
+          case "*" => HttpOriginRange.*
+          case s => HttpOriginRange.Default(HttpOrigin(s) :: Nil)
+        }
+
+      override val corsAllowedOrigins: Option[HttpOriginRange] =
+        (Try(parseMultipleOrigins(rpcConfig.getStringList("cors-allowed-origins").asScala)) recoverWith {
+          case _ => Try(parseSingleOrigin(rpcConfig.getString("cors-allowed-origins")))
+        }).toOption
```

Comment:
It's not immediately clear whether not specifying this setting is equivalent to `cors-allowed-origins = []` or `cors-allowed-origins = *`. BTW, TypesafeConfig authors advise against implementing optional (default) settings.

Also, is `cors-allowed-origins = ""` a valid setting?

Reply:
Well ... `*` is standard for _all_ (`Access-Control-Allow-Origin: *`) and  `[]` refers to no domains (as it's an empty seq). `""` Is not a valid domain so it will fail.

Can you think of a better alternative? I'm open to suggestions

Reply:
Yes, but I'm saying that this setting does not need to be provided at all. And it's not clear what that means. IMO, best approach is avoiding defaults, so instead of `.toOption` just let it fail.

Reply:
I've just removed the optional param. Btw, while taking a look at this I've found that if there is an error while loading RPC Server, the [app will start besides that](https://github.com/input-output-hk/mantis/blob/phase/daedalus/src/main/scala/io/iohk/ethereum/App.scala#L41). Maybe its a good idea to change that too, wdyt?

Reply:
I agree, the client should not start. If you feel it's OK to change it on this PR, then you have my blessing 😉